app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/CheckYourAnswersController.scala

```diff
@@ -72,17 +74,34 @@ class CheckYourAnswersController @Inject()(override val controllerComponents: Me
     }
   }
 
-  private def declarationConfirmation(request: DeclarationJourneyRequest[AnyContent], goods: DeclarationGoods)
-                                     (implicit headerCarrier: HeaderCarrier): Future[Result] = {
+  private def declarationConfirmation(declaration: Declaration)
+                                     (implicit request: DeclarationJourneyRequest[AnyContent]): Future[Result] =
     request.declarationJourney.declarationType match {
-      case Export => Future.successful(Redirect(routes.DeclarationConfirmationController.onPageLoad()))
-      case Import => makePayment(goods).map(res => Redirect(connector.extractUrl(res).nextUrl.value))
+      case Export => mibConnector.persistDeclaration(declaration).flatMap(resetAndRedirect)
+      case Import => processImportDeclaration(declaration)
     }
-  }
+
+  private def processImportDeclaration(declaration: Declaration)
+                                      (implicit request: DeclarationJourneyRequest[AnyContent]): Future[Result] =
+    for {
+      persist <- mibConnector.persistDeclaration(declaration)
+      pay     <- makePayment(declaration.declarationGoods)
+      _       <- resetJourney(persist)
+    } yield Redirect(connector.extractUrl(pay).nextUrl.value)
 
   private def makePayment(goods: DeclarationGoods)(implicit headerCarrier: HeaderCarrier): Future[HttpResponse] =
```

Comment:
maybe 'createPaymentSession' or similar rather than 'makePayment'? This call doesn't actually make the payment so might mislead the unwary.

Reply:
done

Reply:
Not quite - I also meant re-name this controller method makePayment() too.

Reply:
done